### ISSUE TITLE
Add a way to get a function's description from a shell script

### ISSUE THREAD

**Issue by adambrenecki** [fish-shell/fish-shell#597]
Currently, the only way to do this is (as far as I can tell) is by parsing the full output of `functions` to pull out the argument to `--description` (eg `functions $func_name | head -n 1 | sed -E "s|.*'(.*)'.*|\1|"`).

**Comment by jonmarkprice** [comment]
:+1:

**Comment by 8tt** [comment]
Would be very useful for scripts replacing simple Make files.

**Comment by floam** [comment]
It's terribly unpleasant what it does, but you might find `__fish_describe_command` useful.

**Comment by floam** [comment]
OK, I'm surprised, that doesn't do what I imagined it would after actually looking at it. Nevermind!

**Comment by floam** [comment]
Probably if you give `functions -d` one argument it should output the description.

**Comment by floam** [comment]
interesting: `functions -d foo any_command_with_a_description` - doesn't actually seem to affect the description shown in the pager for me.

**Comment by floam** [comment]
What description should `fish_indent` have?

```
$ functions fish_indent
function fish_indent --description 'Indenter and prettifier for fish code'
...
```

```
$ complete -Cfish_indent
fish_indent	Executable, 1.4MB
```

seen in pager for `fish_<TAB>`:
```
fish_indent                         (-- indenter and prettifier)
```

**Comment by krader1961** [comment]
Introspection is a useful tool. This is similar to issue #3295 and should be implemented just on general principal. Having said that I'm wondering what the use case is for this feature.

**Comment by adambrenecki** [comment]
Virtualfish has a single `vf` command with a bunch of subcommands. Internally, the subcommands are all functions with names like `__vf_x` (since fish doesn't have namespaces or private functions), and we parse out the function description to display help text.

**Comment by krader1961** [comment]
We now support `functions --metadata --verbose $func_name`. It currently does not report the function description. Adding the description to its output is an obvious enhancement. Presumably it should use the same encoding/escaping mechanism that `type $func_name` uses to deal with descriptions that contain newlines or other special characters. As in this example:

```
$ function wtf --description '[]{}$abc
      def'
      echo yes
  end
$ type wtf
wtf is a function with definition
function wtf --description \[\]\{\}\$abc\ndef
    echo yes
end
```

A pull-request to implement that would be welcomed.

**Comment by krader1961** [comment]
I have a proof of concept implementation that addresses this issue by building on the `functions --metadata` implementation I implemented the fix for issue #3295 a month ago. I need to add some unit tests but expect to merge this in the next few days.
